lib/routes/validators.js

```diff
@@ -134,40 +134,67 @@ module.exports.isValidEmailAddress = function(value) {
 }
 
 module.exports.redirectTo = function redirectTo(base) {
-  const redirectTo = isA.string().max(512)
+  const validator = isA.string().max(512)
   let hostnameRegex = null
   if (base) {
     hostnameRegex = new RegExp('(?:\\.|^)' + base.replace('.', '\\.') + '$')
   }
-  redirectTo._tests.push(
+  validator._tests.push(
     {
       func: (value, state, options) => {
         if (value !== undefined && value !== null) {
-          const normalizedValue = module.exports.isValidUrl(value, hostnameRegex)
-          if (normalizedValue) {
-            return normalizedValue
+          if (module.exports.isValidUrl(value, hostnameRegex)) {
+            return value
           }
         }
 
-        return redirectTo.createError('string.base', { value }, state, options)
+        return validator.createError('string.base', { value }, state, options)
       }
     }
   )
-  return redirectTo
+  return validator
 }
 
-module.exports.isValidUrl = function (redirect, hostnameRegex) {
-  var parsed = url.parse(redirect)
+module.exports.url = function url(options) {
+  const validator = isA.string().max(512).uri(options)
+  validator._tests.push(
+    {
+      func: (value, state, options) => {
+        if (value !== undefined && value !== null) {
+          if (module.exports.isValidUrl(value)) {
+            return value
+          }
+        }
+
+        return validator.createError('string.base', { value }, state, options)
+      }
+    }
+  )
+  return validator
+}
+
+module.exports.isValidUrl = function (url, hostnameRegex) {
+  let parsed
+  try {
+    parsed = new URL(url)
```

Comment:
I switched to the new `URL` class because it does the right thing with unescaped special characers:

```
> const { URL } = require('url')
> new URL('https://test\u010Dme.com').href
'https://xn--testme-l2a.com/'
> new URL('https://test.me/special/\u010D/chars').href
'https://test.me/special/%C4%8D/chars'
```